### PR TITLE
Kolomiets branch - osmand and bugfixing (search filter)

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java [hunk 1]
```diff
@@ -142,19 +145,24 @@ public MapActivityActions(MapActivity mapActivity) {
 		settings = mapActivity.getMyApplication().getSettings();
 		routingHelper = mapActivity.getMyApplication().getRoutingHelper();
 		drawerLogoHeader = new ImageView(mapActivity);
-		drawerLogoHeader.setPadding(-AndroidUtils.dpToPx(mapActivity, 8f), AndroidUtils.dpToPx(mapActivity, 16f), 0,0);
-		drawerOsmAndFooter = mapActivity.getLayoutInflater().inflate(R.layout.powered_by_osmand_item, null);
+		drawerLogoHeader
+				.setPadding(-AndroidUtils.dpToPx(mapActivity, 8f), AndroidUtils.dpToPx(mapActivity, 16f), 0,
+						0);
+		drawerOsmAndFooter = mapActivity.getLayoutInflater()
+				.inflate(R.layout.powered_by_osmand_item, null);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java]: Redundant line feed

File: OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java [hunk 2]
```diff
@@ -142,19 +145,24 @@ public MapActivityActions(MapActivity mapActivity) {
...
 	}
 
 	public void addAsTarget(double latitude, double longitude, PointDescription pd) {
 		TargetPointsHelper targets = getMyApplication().getTargetPointsHelper();
-		targets.navigateToPoint(new LatLon(latitude, longitude), true, targets.getIntermediatePoints().size() + 1,
+		targets.navigateToPoint(new LatLon(latitude, longitude), true,
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java]: Why these many extra line feeds added? You may do that in new code, but do not touch existing.

File: OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java [hunk 3]
```diff
@@ -40,341 +43,378 @@
...
+
+  public boolean isWidgetAvailable(@NonNull String key, ApplicationMode appMode) {
+    Set<ApplicationMode> set = widgetsAvailabilityMap.get(key);
+    if (set == null) {
+      return true;
+    }
+    return set.contains(appMode);
+  }
+
+  public boolean setNavDrawerLogoWithParams(String uri, @Nullable String packageName,
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java]: @Nullable String uri?

File: OsmAnd/src/net/osmand/plus/activities/actions/OsmandRestoreOrExitDialog.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package net.osmand.plus.activities.actions;
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/activities/actions/OsmandRestoreOrExitDialog.java]: Spaces

File: OsmAnd/AndroidManifest.xml [hunk 5]
```diff
@@ -157,30 +157,35 @@
...
-				<category android:name="android.intent.category.BROWSABLE" />
-				<data android:scheme="file"/>
-				<data android:host="*"/>
-				<data android:pathPattern=".*\\.kml" />
-				<data android:pathPattern=".*\\..*\\.kml" />
-				<data android:pathPattern=".*\\..*\\..*\\.kml" />
-				<data android:pathPattern=".*\\..*\\..*\\..*\\.kml" />
-				<data android:pathPattern=".*\\..*\\..*\\..*\\..*\\.kml" />
-			</intent-filter>
+			<!--<intent-filter-->
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/AndroidManifest.xml]: +

File: OsmAnd/res/layout/main.xml [hunk 6]
```diff
@@ -97,23 +99,25 @@
         android:id="@+id/drawer_relative_layout"
         android:layout_width="280dp"
         android:layout_height="match_parent"
-        android:layout_gravity="start">
+        android:layout_gravity="start"
+        >
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/res/layout/main.xml]: +

File: OsmAnd/res/layout/simple_list_menu_item.xml [hunk 7]
```diff
@@ -5,17 +5,19 @@
               android:layout_height="@dimen/list_item_height">
 
     <TextView
+        android:layoutDirection="locale"
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/res/layout/simple_list_menu_item.xml]: removed, it didn't needed

File: OsmAnd/src/net/osmand/aidl/OsmandAidlApi.java [hunk 8]
```diff
@@ -1641,6 +1646,50 @@ public void onFinish(AppInitializer init) {
...
+          }
+        }
+      });
+    } else {
+      callback.onAppInitialized();
+    }
+    return true;
+  }
+
+//	void registerForOsmandInitialization(final IOsmAndAidlCallback callback) throws RemoteException {
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/aidl/OsmandAidlApi.java]: +

File: OsmAnd/src/net/osmand/aidl/navdrawer/NavDrawerFooterParams.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
+package net.osmand.aidl.navdrawer;
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/aidl/navdrawer/NavDrawerFooterParams.java]: +

File: OsmAnd/src/net/osmand/aidl/OsmandAidlService.java [hunk 10]
```diff
@@ -71,356 +75,363 @@
...
-			}
-			return false;
-		}
-
-		@Override
-		public boolean calculateRoute(CalculateRouteParams params) throws RemoteException {
-			if (params == null || params.getEndPoint() == null) {
-				return false;
-			} else {
+  private static final Log LOG = PlatformUtil.getLog(OsmandAidlService.class);
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/aidl/OsmandAidlService.java]: +

File: OsmAnd/src/net/osmand/aidl/navdrawer/NavDrawerHeaderParams.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
+package net.osmand.aidl.navdrawer;
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/aidl/navdrawer/NavDrawerHeaderParams.java]: +

File: OsmAnd/src/net/osmand/aidl/plugins/PluginParams.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
+package net.osmand.aidl.plugins;
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/aidl/plugins/PluginParams.java]: +

File: OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java [hunk 13]
```diff
@@ -40,341 +43,378 @@
...
-	}
-
-	public void addListener(OsmAndAppCustomizationListener listener) {
-		this.listeners.add(listener);
-	}
-
-	public void removeListener(OsmAndAppCustomizationListener listener) {
-		this.listeners.remove(listener);
-	}
+  protected OsmandApplication app;
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java]: +

File: OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java]: +

File: OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java [hunk 14]
```diff
@@ -142,19 +145,24 @@ public MapActivityActions(MapActivity mapActivity) {
 		settings = mapActivity.getMyApplication().getSettings();
 		routingHelper = mapActivity.getMyApplication().getRoutingHelper();
 		drawerLogoHeader = new ImageView(mapActivity);
-		drawerLogoHeader.setPadding(-AndroidUtils.dpToPx(mapActivity, 8f), AndroidUtils.dpToPx(mapActivity, 16f), 0,0);
-		drawerOsmAndFooter = mapActivity.getLayoutInflater().inflate(R.layout.powered_by_osmand_item, null);
+		drawerLogoHeader
+				.setPadding(-AndroidUtils.dpToPx(mapActivity, 8f), AndroidUtils.dpToPx(mapActivity, 16f), 0,
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java]: +

File: OsmAnd/src/net/osmand/plus/activities/actions/OsmandRestoreOrExitDialog.java (hunk 4 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/activities/actions/OsmandRestoreOrExitDialog.java]: +

File: OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java (hunk 3 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/OsmAndAppCustomization.java]: +

**Pull request merged**